src/client/pages/Afis/useAfisThemaData.hook.tsx

```diff
@@ -0,0 +1,189 @@
+          factuur.documentDownloadLink,
+          `factuur ${factuur.factuurNummer}`
+        )}
+      />
+    ) : (
+      factuur.factuurNummer
+    ),
+  };
+}
+
+function toGenericDocument(
+  document: GenericDocument | string,
+  title: string
+): GenericDocument {
+  if (typeof document === 'string') {
+    return {
+      id: document,
+      url: document,
+      title,
+      datePublished: new Date().toISOString(),
+    };
+  }
+  return document;
+}
+
+const businessPartnerDetailsLabels: DisplayProps<AfisBusinessPartnerDetailsTransformed> =
+  {
+    fullName: 'Debiteurnaam',
+    businessPartnerId: 'Debiteurnummer',
+    email: 'E-mailadres factuur',
+    phone: 'Telefoonnummer',
+  };
+
+export function useAfisBetaalVoorkeurenData(
+  businessPartnerIdEncrypted: AfisBusinessPartnerKnownResponse['businessPartnerIdEncrypted']
+) {
+  const [
+    businesspartnerDetails,
+    businesspartnerDetailsApi,
+    fetchBusinessPartner,
+  ] = useAppStateBagApi<AfisBusinessPartnerDetailsTransformed | null>({
+    bagThema: BagThemas.AFIS,
+    key: `${businessPartnerIdEncrypted}-betaalvoorkeuren`,
+  });
+
+  useEffect(() => {
+    if (businessPartnerIdEncrypted) {
+      fetchBusinessPartner({
+        url: `${BFFApiUrls.AFIS_BUSINESSPARTNER}/${businessPartnerIdEncrypted}`,
+      });
+    }
+  }, [businessPartnerIdEncrypted, fetchBusinessPartner]);
+
+  return {
+    businesspartnerDetails,
+    businessPartnerDetailsLabels,
+    isLoadingBusinessPartnerDetails: businesspartnerDetailsApi.isLoading,
+    hasBusinessPartnerDetailsError: isError(
+      businesspartnerDetailsApi.data,
+      false
+    ),
+    hasEmandatesError: false,
+    hasFailedEmailDependency: hasFailedDependency(
+      businesspartnerDetailsApi.data,
+      'email'
+    ),
+    hasFailedPhoneDependency: hasFailedDependency(
+      businesspartnerDetailsApi.data,
+      'phone'
+    ),
+    eMandateTableConfig,
+    eMandates: [],
+    isLoadingEmandates: false,
+  };
+}
+
+function getInvoiceStatusDescription(factuur: AfisFactuur): ReactNode {
+  switch (factuur.status) {
+    case 'openstaand':
+      return (
+        <>
+          Openstaand:{' '}
+          <MaRouterLink maVariant="fatNoUnderline" href={factuur.paylink ?? ''}>
+            {factuur.amountOwedFormatted} betaal nu
+          </MaRouterLink>
+        </>
+      );
+    case 'in-dispuut':
+      return 'In dispuut';
+    case 'gedeeltelijke-betaling':
+      return `Automatische incasso - Betaal het openstaande bedrag van ${factuur.amountOwedFormatted} via bankoverschrijving`;
+    case 'betaald':
+      return `Betaald ${factuur.debtClearingDateFormatted ? `op ${factuur.debtClearingDateFormatted}` : ''}`;
+    case 'automatische-incasso':
+      return (
+        <>
+          {factuur.amountOwedFormatted} wordt automatisch van uw <br />
+          rekening afgeschreven
+        </>
+      );
+    default:
+      return capitalizeFirstLetter(factuur.status ?? '');
+  }
+}
```

Comment:
Ik denk dat we dit beter naar de BFF kunnen verplaatsen, ik wil de business logica zoveel mogelijk daar laten. We kunnen een additional prop maken in AfisFactuur. `statusDescription` oid.  Dan hoef je alleen nog die openstaande prop te veranderen naar een MARouterLink

Reply:
helemaal goed, doen we!